backend.native/tests/runtime/workers/worker5.kt

```diff
@@ -19,4 +17,36 @@ import kotlin.native.concurrent.*
     }
     worker.requestTermination().result
     println("OK")
+}
+
+var done = false
+
+@Test fun runTest1() {
+    val worker = Worker.current
+    done = false
+    // Here we request execution of the operation on the current worker.
+    worker.executeAfter(0, {
+        done = true
+    }.freeze())
+    while (!done)
+        worker.processQueue()
+}
+
+// Ensure that termination of current worker on main thread doesn't lead to problems.
+@Test fun runTest2() {
+    val worker = Worker.current
+    val future = worker.requestTermination(false)
+    worker.processQueue()
+    assertEquals(future.state, FutureState.COMPUTED)
+    future.consume {}
+    // After termination request this worker is no longer addressable.
+    assertFailsWith<IllegalStateException> { worker.executeAfter(0, {
+        println("BUG!")
+    }.freeze()) }
+}
+
+fun main() {
```

Comment:
Maybe add `flags = ['-tr']` to `build.gradle` instead?

Reply:
Maybe.

Reply:
Turns out, that this will force to change test output to that of test logger, which doesn't look like best idea.